packages/client/src/impl/QueryApiImpl.ts

```diff
@@ -0,0 +1,78 @@
+import {RecordBatchReader} from 'apache-arrow'
+import QueryApi from '../QueryApi'
+import {Ticket} from '../generated/flight/Flight'
+import {FlightServiceClient} from '../generated/flight/Flight.grpc-client'
+import {ConnectionOptions, QueryType} from '../options'
+import {replaceURLProtocolWithPort} from '../util/fixUrl'
+import * as grpc from '@grpc/grpc-js'
+import {createInt32Uint8Array} from '../util/common'
+
+export default class QueryApiImpl implements QueryApi {
+  private closed = false
+  private flightClient: FlightServiceClient
+
+  constructor(private _options: ConnectionOptions) {
+    const {url, safe} = replaceURLProtocolWithPort(this._options.url)
+    const credentials =
+      grpc.credentials[safe ?? true ? 'createSsl' : 'createInsecure']()
+    this.flightClient = new FlightServiceClient(url, credentials)
+  }
+
+  async *query(
+    query: string,
+    database: string,
+    queryType: QueryType = 'sql'
+  ): AsyncGenerator<Record<string, any>, void, void> {
+    if (this.closed) {
+      throw new Error('queryApi: already closed!')
+    }
+    const client = this.flightClient
+
+    const ticketData = {
+      database: database,
+      sql_query: query,
+      query_type: queryType,
+    }
+    const ticket = Ticket.create({
+      ticket: new TextEncoder().encode(JSON.stringify(ticketData)),
+    })
+
+    const metadata = new grpc.Metadata()
+    const token = this._options.token
+    if (token) metadata.set('authorization', 'Bearer ' + token)
```

Comment:
Please use template string:

```suggestion
    if (token) metadata.set('authorization', `Bearer ${token}`)
```